packages/ampd-sdk/src/event/event_handler.rs

```diff
@@ -0,0 +1,520 @@
+use std::fmt::Display;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use cosmrs::Any;
+use error_stack::{report, Context, Report, ResultExt};
+use events::{AbciEventTypeFilter, Event};
+use futures::{pin_mut, Stream, TryFutureExt};
+use mockall::automock;
+use report::ErrorExt;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tokio::time::interval;
+use tokio_stream::{Elapsed, StreamExt};
+use tokio_util::sync::CancellationToken;
+use tracing::{error, info};
+use typed_builder::TypedBuilder;
+
+use crate::event::callbacks::{
+    default_broadcast_error_cb, default_event_subscription_error_cb, default_handler_error_cb,
+};
+use crate::future::{with_retry, RetryPolicy};
+use crate::grpc::client;
+
+#[automock(
+    type Err = Error;
+    type Event = Event;
+)]
+#[async_trait]
+pub trait EventHandler: Send + Sync {
+    type Err: Context;
+    type Event: TryFrom<Event>;
+
+    async fn handle(
+        &self,
+        event: &Self::Event,
+        token: &CancellationToken,
+    ) -> Result<Vec<Any>, Self::Err>;
```

Comment:
This should be `error_stack::Result`.
```suggestion
    use error_stack::Result;
    
    ...

    ) -> Result<Vec<Any>, Self::Err>;
```